src/main/java/teammates/ui/controller/ImageUploadAction.java

```diff
@@ -0,0 +1,99 @@
+package teammates.ui.controller;
+
+import java.util.List;
+import java.util.Map;
+
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreFailureException;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+public class ImageUploadAction extends Action {
+
+    FileUploadPageData data;
+
+    @Override
+    protected ActionResult execute() {
```

Comment:
Both methods are more than 50% similar. Is it absolutely necessary for those similarity to be duplicated in both `ImageUploadAction` and its admin email counterpart? Ideally (albeit may not be possible) `ImageUploadAction` contains just the method definition of `execute` with appropriate parts to be overridden by its admin email counterpart.